Fixes plasma grenades

OuterMostMonk opened the pull request:
:cl: OuterMostMonk
tweak: Plasma grenades explode when thrown
tweak: Martyrdom grenades work again
/:cl:

Plasma grenades will now explode even if they do not embed in a target. Martyrdom grenades will also explode properly when dropped/thrown as intended.

OuterMostMonk committed:
Fixes plasma grenades

Plasma grenades will now explode even if they do not embed in a target. Martyrdom grenades will now also explode properly when dropped/thrown as intended.

X0-11 commented:
Oh, *that's* why they weren't activating. Thanks for catching this.

X0-11 commented on code/modules/halo/weapons/covenant/grenades.dm:
```diff
@@ -55,6 +55,7 @@
 			h.update_inv_r_hand()
 
 /obj/item/weapon/grenade/plasma/throw_impact(var/atom/A)
+	start_timer()
```
This should probably go through the usual checks throw impact does before starting the timer.

OuterMostMonk committed:
Moves start_timer()

This was a real giga brain code moment on my part. Grenades would explode when thrown even if not activated lol. Now it will start the timer only if thrown while activated.

OuterMostMonk commented:
Yeah that start_timer() was placed funny (exploding inactivated grenades when thrown), now it should work fine. Can't place it after the checks later as that would cause it to only explode if it hits a person.
